packages/block-library/src/heading/heading-checker.js

```diff
@@ -0,0 +1,58 @@
+/**
+ * WordPress dependencies
+ */
+import { speak } from '@wordpress/a11y';
+import { __ } from '@wordpress/i18n';
+import { Notice } from '@wordpress/components';
+import { useEffect } from '@wordpress/element';
+import { compose } from '@wordpress/compose';
+import { withSelect } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import { computeOutlineHeadings } from '../../../editor/src/components/document-outline/index.js';
+
+export const HeadingChecker = ( { blocks = [], selectedLevel, selectedHeadingId } ) => {
+	const headings = computeOutlineHeadings( blocks );
+
+	// Iterate headings to find prevHeadingLevel
+	let prevHeadingLevel = 1;
+	let i = 0;
+	for ( i = 0; i < headings.length; i++ ) {
+		if ( i >= 1 && headings[ i ].clientId === selectedHeadingId ) {
+			prevHeadingLevel = headings[ i - 1 ].level;
+		}
+	}
+
+	const isIncorrectLevel = ( selectedLevel === 1 || selectedLevel > prevHeadingLevel + 1 );
```

Comment:
`selectedLevel` can be read from the currently selected block. We don't need to pass it down.

Reply:
Solve it